kaldb/src/main/java/com/slack/kaldb/logstore/search/KaldbDistributedQueryService.java

```diff
@@ -133,8 +134,10 @@ private void updateStubs() {
     span.tag("queryServerCount", String.valueOf(stubs.size()));
 
     for (KaldbServiceGrpc.KaldbServiceFutureStub stub : stubs.values()) {
+      // make sure all underlying futures finish executing (successful/cancelled/failed/other)
+      // and cannot be pending when the successfulAsList.get(SAME_TIMEOUT_MS) runs
       ListenableFuture<KaldbSearch.SearchResult> searchRequest =
-          stub.withDeadlineAfter(READ_TIMEOUT_MS, TimeUnit.MILLISECONDS)
+          stub.withDeadlineAfter(GRPC_QUERY_TIMEOUT_DURATION.toMillis(), TimeUnit.MILLISECONDS)
```

Comment:
This may not work as expected - since this is calculated once as a static value, any test that dynamically updates the `READ_TIMEOUT_MS` will not be reflected here.

Reply:
That is a solid point...